### PR TITLE
Change rules of given prioritization test

### PR DESCRIPTION
**Pull request by odersky**
This is the new rules, but with a small tweak and from 3.5 so that we can test open CB. 

The tweak is that we compare according to new rules only if both side are givens, rather than implicits.

### CODE REVIEW

**Comment by odersky** [comment]
@WojciechMazur Could you run open CB on this? I want to understand why we got do different results in the two previous versions, so I rebased every version on current main to be able to compare better.

File: compiler/src/dotty/tools/dotc/typer/Applications.scala [hunk 1]
```diff
@@ -1680,27 +1686,28 @@ trait Applications extends Compatibility {
    *  an alternative that takes more implicit parameters wins over one
    *  that takes fewer.
    */
-  def compare(alt1: TermRef, alt2: TermRef)(using Context): Int = trace(i"compare($alt1, $alt2)", overload) {
+  def compare(alt1: TermRef, alt2: TermRef, preferGeneral: Boolean = false)(using Context): Int = trace(i"compare($alt1, $alt2)", overload) {
     record("resolveOverloaded.compare")
 
-    /** Is alternative `alt1` with type `tp1` as specific as alternative
+    val compareGivens = alt1.symbol.is(Given) && alt2.symbol.is(Given)
```
Comment (odersky) [inline compiler/src/dotty/tools/dotc/typer/Applications.scala:1692]:
```suggestion
    val compareGivens = true // alt1.symbol.is(Given) && alt2.symbol.is(Given)
```

**Comment by odersky** [comment]
Can we get another open CB run with the latest commit?

**Comment by WojciechMazur** [comment]
When compiled from [aa5686e](https://github.com/scala/scala3/pull/20277/commits/aa5686e8b75e0443f6d4d5ed3bd8ef3ea52064af) we can find at least 5 build failures. We've tested 1613 projects - 52 of these are already failing on current main branch (latest nightly)

| Project | Version | Build URL | Notes |
| ------- | ------- | --------- | ----- |
| guymers/foobie | 0.14.8 | [Open CB logs](https://github.com/VirtusLab/community-build3/actions/runs/8877001132/job/24370062008) | Ambiguous given instances |
| kierendavies/skunk-migrate | 0.3.0 | [Open CB logs](https://github.com/VirtusLab/community-build3/actions/runs/8877001132/job/24370081641) | Ambiguous given instances |
| lichess-org/lila-ws | HEAD | [Open CB logs](https://github.com/VirtusLab/community-build3/actions/runs/8876998279/job/24370033856) | implicit conversions cannot be applied because they are ambiguous |
| minosiants/pencil | 2.1.0 | [Open CB logs](https://github.com/VirtusLab/community-build3/actions/runs/8876998279/job/24376225683) | Ambiguous given instances |
| torrentdam/bittorrent | 3.1.2 | [Open CB logs](https://github.com/VirtusLab/community-build3/actions/runs/8876998279/job/24376681808) | implicit conversions cannot be applied because they are ambiguous |

**Comment by odersky** [comment]
Thanks for the analysis! So we can summarize as follows:

 - If we switch to the new prioritization rules for givens, there are no regressions. 
 - If at some point in the future all CB projects convert from implicits to givens, there are the five projects above that will need to be adapted to the new priority rules. That seems not too bad.

So I think we can roll out the change as planned. I'll drop the last test commit.

**Pull request closed without merging**